### ISSUE TITLE
AutoTokenizer in Convert_vima notebook

### ISSUE THREAD

**Issue by ManishGovind** [LostXine/LLaRA#10]
Hello @LostXine ,

When i try to generate instruction Tuning BC (step 4). I get the below error :
```


from transformers import AutoTokenizer
      [2](vscode-notebook-cell:?execution_count=4&line=2) import torch 
----> [3](vscode-notebook-cell:?execution_count=4&line=3) tokenizer = AutoTokenizer.from_pretrained("llava-hf/llava-1.5-7b-hf")
      [4](vscode-notebook-cell:?execution_count=4&line=4) print(tokenizer.vocab_size)
      [6](vscode-notebook-cell:?execution_count=4&line=6) # this is for RT-2
.......

[110](https://vscode-remote+ssh-002dremote-002bhpc-002echarlotte-002eedu.vscode-resource.vscode-cdn.net/users/mgovind/LLaRA/datasets/~/miniconda3/envs/llara/lib/python3.10/site-packages/transformers/tokenization_utils_fast.py:110)     # We have a serialization from tokenizers which let us directly build the backend
--> [111](https://vscode-remote+ssh-002dremote-002bhpc-002echarlotte-002eedu.vscode-resource.vscode-cdn.net/users/mgovind/LLaRA/datasets/~/miniconda3/envs/llara/lib/python3.10/site-packages/transformers/tokenization_utils_fast.py:111)     fast_tokenizer = TokenizerFast.from_file(fast_tokenizer_file)

Exception: data did not match any variant of untagged enum ModelWrapper at line 277156 column 3
```

May i know what should be the transformers version. I followed the same instructions for setting up LLaRA according to  README.md . Thanks for your wonderful work !!

**Comment by LostXine** [comment]
Hi @ManishGovind ,

Thanks for your interest in our work. We are using "transformers==4.37.2", "tokenizers==0.15.1" as suggested by the original LLaVA project here https://github.com/LostXine/LLaRA/blob/f60f59d440b8f5b122464c169c962614c272b330/train-llava/pyproject.toml#L17

Could you help me confirm this?

Thanks,

Xiang

**Comment by ManishGovind** [comment]
Yes, I'm also using the same. 

Name: transformers
Version: 4.37.2
Summary: State-of-the-art Machine Learning for JAX, PyTorch and TensorFlow
Home-page: https://github.com/huggingface/transformers
Author: The Hugging Face team (past and future) with the help of all our contributors (https://github.com/huggingface/transformers/graphs/contributors)
Author-email: transformers@huggingface.co
License: Apache 2.0 License
Requires: filelock, huggingface-hub, numpy, packaging, pyyaml, regex, requests, safetensors, tokenizers, tqdm
Required-by: #N/A
(llara) pip show tokenizers
Name: tokenizers
Version: 0.15.1
Summary: 
Home-page: https://github.com/huggingface/tokenizers
Author: Anthony MOI <m.anthony.moi@gmail.com>
Author-email: Nicolas Patry <patry.nicolas@protonmail.com>, Anthony Moi <anthony@huggingface.co>
License: 
Requires: huggingface_hub
Required-by: #N/A

**Comment by LostXine** [comment]
I see, let me initialize a new environment and test it again. Thanks for bringing this to my attention.

**Comment by ManishGovind** [comment]
Sure, No problem. Looking forward to your reply.

**Comment by LostXine** [comment]
Hi @ManishGovind ,

I've reproduced the issue and found a temporary fix. You can bypass the current transformers package version requirement and upgrade to the latest version with the following command:
```
pip install -U transformers
```
After the upgrade, the tokenizer should work as expected. However, I haven't tested compatibility with other parts of the code, so if any issues arise, you may need to revert to version 4.37.2.

I'll implement this quick fix for now and work on a proper solution.

Thanks for your understanding.

Best,

Xiang

**Comment by ManishGovind** [comment]
I will also try to upgrade the version and see if it works.
Thanks for your help !!!

Best,
Manish

**Comment by ManishGovind** [comment]
Hi @LostXine ,

I Re-trained llara with  D-InBC-Aux-D-80K instruction data and wanted to reproduce the results. But i end up with these results. 

May i know what could be the issue?


<img width="791" alt="Image" src="https://github.com/user-attachments/assets/223096b5-6545-4dd3-87be-4d76499b8655" />

Thanks,
Manish

**Comment by LostXine** [comment]
Hi @ManishGovind , 

I could not find `D-InBC-Aux-D-80K` in the image you posted. Could you provide more context? 

Thanks,

**Comment by ManishGovind** [comment]
so the first two rows are  nothing but the D-inBc+Aux-D. I have used D-inBC-text-multi-train-80k-front.json(i.e, D-inBC + Auxiliary tasks) to train . Do you want me to share  my inference json ?

**Comment by LostXine** [comment]
@ManishGovind Thanks for the clarification, yes please share whatever you think related, e.g. training data, inference json. Also, could you also share the training details?

We have people replicated the results before so I think we can figure it out (see https://github.com/LostXine/LLaRA/issues/3)

**Comment by ManishGovind** [comment]
Hi @LostXine,

I noticed that the `D-InBC-Aux-D-80K `checkpoints provided  in the huggingface were trained for 8 epochs. However, the training script I followed, which is for `D-InBC-Aux-D-0d8K,` has a total of 2 epochs. Using the provided pretrained checkpoints, I was able to reproduce the results.

Based on this, I believe I need to train for more epochs with a batch size of 128 to fully replicate the results.

For your reference, here are my training and inference configurations:

[trainer_state.json](https://github.com/user-attachments/files/18831501/trainer_state.json)
[\[hso\]D-inBC-AuxB-VIMA-80k.json](https://github.com/user-attachments/files/18831500/hso.D-inBC-AuxB-VIMA-80k.json)


Best,
Manish

**Comment by LostXine** [comment]
Hi @ManishGovind ,

Thank you for sharing these files. I appreciate your understanding on the fact that we have not fully iterated through all the configurations in this repository. However, the necessary training scripts should be available within the repository, along with the information in Tables 21-27. Based on the results in Table 26 (Page 37), using 2 epochs should yield reasonable results, although we have not tested a smaller batch size on the 80k scale datasets.

The trainer_state.json you shared has a smaller `global_step` than I expected even considering the smaller epoch you were using. To assist further, I would like to verify the training data you have been using. Additionally, could you please clarify the naming convention? In the original paper, `Aux-B` refers to the use of the following four datasets (please ensure to select the corresponding dataset scale): `aux-obj_det-train-0d8k-front.json`, `aux-obj_loc-train-0d8k-front.json`, `aux-act_pred-train-0d8k-front.json`, and `aux-fut_pred-train-0d8k-front.json`. On the other hand, `Aux-D` refers to using all six auxiliary datasets.

Thank you again.

Best,
Xiang

**Comment by ManishGovind** [comment]
Thanks for clarifying Aux-B and Aux-D. After checking thoroughly , I just wanted to confirm that I am using D-inBC-80K as the training set without any auxiliary tasks. So, the results I previously shared do not include any auxiliary data, which might also be a contributing factor.

Additionally, could you please clarify few doubts :

1)  which object detector was used in the D-inBC+OD method? I understand that for the real-world robot experiment, you mentioned using Owlv2 as the object detector.

2) Also, in the ratio of auxiliary dataset mixture with inBC data, what does ‘*’ signify in the following image?

<img width="527" alt="Image" src="https://github.com/user-attachments/assets/b794b240-ba4c-4688-9169-06ace2db3a12" /> 

 This image refers to Table 24 in the paper.

Since I now understand that the issue might be related to hyperparameters and the correct composition of auxiliary data, I will rerun the experiment while taking these factors into consideration.

I really appreciate for your help!

Best,
Manish

**Comment by LostXine** [comment]
@ManishGovind thanks for the update and I'm looking forward to your further results.

Answer to your questions:
1. The object detector is a modified mask RCNN, the implementation can be found here: https://github.com/LostXine/LLaRA/tree/main/train-maskrcnn We have a description in detail on object detection in appendix A3, page 18.
2. The `*` means  the reference images that appeared only in the task description are not used to generate this auxiliary dataset. This note should have been included in the title of this table. Please check the latest version of our paper if you could not find it: https://arxiv.org/pdf/2406.20095

Best,
Xiang

**Comment by ManishGovind** [comment]
@LostXine Great Thank you !! I will look into the latest version.

Best,
Manish
